gelf/tcpwriter.go

```diff
@@ -87,3 +63,19 @@ func (w *TCPWriter) Write(p []byte) (n int, err error) {
 
 	return len(p), nil
 }
+
+func (w *TCPWriter) writeToSocketWithReconnectAttempts(zBytes []byte) (n int, err error) {
+	var errConn error
+	w.mu.Lock()
+	for i := 0; n <= 3; i++ {
```

Comment:
Why does this loop always run three times? What happens when the first write is successful? The No. of re-attempts should be configurable. And a reconnect should be logged.